packages/strapi-generate-new/lib/before.js

```diff
@@ -45,8 +48,246 @@ module.exports = (scope, cb) => {
+
+            resolve();
+          });
+        }),
+        new Promise(resolve => {
+          let cmd = `npm install --prefix ${scope.rootPath}_ ${scope.client.connector}@alpha`;
+          if (scope.client.module) {
+            cmd += ` ${scope.client.module}`;
+          }
+
+          exec(cmd, () => {
+            if (scope.client.module) {
+              const lock = require(`${scope.rootPath}_/node_modules/${scope.client.module}/package.json`);
+              scope.client.version = lock.version;
+            }
+
+            resolve();
+          });
+        })
+      ];
+
+      Promise.all(asyncFn)
+      .then(() => {
+        if (scope.client.connector === 'strapi-bookshelf') {
+          const knex  = require(`${scope.rootPath}_/node_modules/knex`)({
+            client: scope.client.module,
+            connection: scope.database
+          });
+
+          knex.raw('select 1+1 as result').then(() => {
+            logger.info('Database connection is a success!');
+            knex.destroy();
+            execSync(`rm -r ${scope.rootPath}_`);
+
+            logger.info('Copying the dashboard...');
+
+            // Trigger callback with no error to proceed.
+            cb.success();
+          })
+          .catch(() => {
+            logger.warn('Database connection failed!');
+            connectionValidation();
+          });
+        } else if (scope.client.connector === 'strapi-mongoose') {
+          const Mongoose = require(`${scope.rootPath}_/node_modules/mongoose`);
+
+          Mongoose.connect(`mongodb://${ (scope.database.username && scope.database.password) ? `${scope.database.username}:${scope.database.password}@` : '' }${scope.database.host}:${scope.database.port}/${scope.database.database}`, {
+            useMongoClient: true
+          }, function (err) {
+            if (err) {
+              logger.warn('Database connection failed!');
+              return connectionValidation();
+            }
+
+            logger.info('Database connection is a success!');
+
+            Mongoose.connection.close();
+
+            execSync(`rm -r ${scope.rootPath}_`);
+
+            logger.info('Copying the dashboard...');
+
+            // Trigger callback with no error to proceed.
+            cb.success();
+          });
+        } else if (scope.client.connector === 'strapi-redis') {
+          const Redis = require(`${scope.rootPath}_/node_modules/ioredis`);
+          const redis = new Redis({
+            port: scope.database.port,
+            host: scope.database.host,
+            password: scope.database.password,
+            db: scope.database.database
+          });
+
+          redis.connect((err) => {
+            if (err) {
+              logger.warn('Database connection failed!');
+              redis.disconnect();
+              return connectionValidation();
+            }
+
+            logger.info('Database connection is a success!');
+
+            redis.disconnect();
```

Comment:
In every case, you have to call `.disconnect()`, you could put this line right before the if condition and remove the lien 265 by the same way.